services-js/registry-certs/client/birth/components/SupportingDocumentsInput.tsx

```diff
@@ -0,0 +1,276 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import {
+  CloseButton,
+  FOCUS_INDICATOR_COLOR,
+  VISUALLYHIDDEN,
+} from '@cityofboston/react-fleet';
+
+interface Props {
+  name: string;
+  title: string;
+  fileTypes?: string[] | string;
+  sizeLimit: FileSize;
+  handleChange(documents: File[]): void;
+}
+
+interface State {
+  selectedFiles: File[];
+  isFocused: boolean;
+}
+
+interface FileSize {
+  amount: number;
+  unit: 'B' | 'KB' | 'MB' | 'GB';
+}
+
+/**
+ * Component that allows a user to select one or more files from their computer.
+ *
+ * Selected files can be individually removed, and additional files can be
+ * added to the group of already-selected files. Duplicate files are identified
+ * by filename, and will be discarded.
+ *
+ * Defaults to accept any file type or size. To restrict the types of files to
+ * accept, assign a string or array of strings to “fileTypes” that contains the
+ * allowed MIME type(s).
+ *
+ * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#Limiting_accepted_file_types
+ *
+ * To disallow large files, pass an object to “sizeLimit” to define the limit:
+ *
+ * { amount: 20, unit: 'MB' }
+ */
+export default class SupportingDocumentsInput extends React.Component<
+  Props,
+  State
+> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      selectedFiles: [],
+      isFocused: false,
+    };
+  }
+
+  private readonly inputRef = React.createRef<HTMLInputElement>();
+
+  // Check to see if a file does not exceed the file size limit, if defined.
+  private checkFileSize = (file: File): boolean | void => {
+    const sizeLimit = handleBytes.convert(this.props.sizeLimit);
+    const fileSize = handleBytes.format(file.size);
+
+    if (file.size > sizeLimit) {
+      // todo: consider handling via callback: https://github.com/CityOfBoston/digital/pull/41#discussion_r218545231
+
+      alert(
+        `There’s a ${this.props.sizeLimit.amount}${
+          this.props.sizeLimit.unit
+        } size limit for documents, but
+${file.name} is ${fileSize.amount.toFixed(2) +
+          fileSize.unit}. Try a different file.`
+      );
+
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  // Add or remove focus styling for the label element.
+  private setFocus = (isFocused: boolean): void => {
+    this.setState({ isFocused });
+  };
+
+  private updateFiles = (files: FileList): void => {
+    const { selectedFiles } = this.state;
+
+    // Create an array from the FileList.
+    const fileArray: File[] = [];
+
+    for (let i = 0; i < files.length; i++) {
+      // Do not allow duplicate filenames, or files that are too large.
+      if (isUnique(files[i]) && this.checkFileSize(files[i])) {
```

Comment:
This doesn't de-dupe file names that get added at the same time.

Reply:
There really isn't a way to add two files with the same name at one time with the system file picker dialog.